src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -2211,7 +2215,7 @@ class RideConstructionWindow final : public Window
 private:
     bool TrackPieceDirectionIsDiagonal(const uint8_t direction)
     {
-        return direction >= 4;
+        return direction & TrackDirectionDiagonalFlag;
```

Comment:
It’s not really a flag. It’s very good you want to get rid of hardcoded number here, so I would do it like this:

```suggestion
        return direction >= NumOrthogonalDirections;
```